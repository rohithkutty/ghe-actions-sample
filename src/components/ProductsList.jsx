import React from 'react';
import Product from './ProductComp';
import './ProductsList.styles.css';

const ProductsList = ({ productsList }) => {
  return (
    <table border='1px' className='products-table'>
      <thead>
        <tr>
          <th>ID</th>
          <th>Product Name</th>
          <th>Quantity</th>
          <th>Price</th>
        </tr>
      </thead>
      <tbody>
        {productsList &&
          productsList.map(({ id, ...remainingProductDetails }) => (
            <Product key={id} id={id} {...remainingProductDetails} />
          ))}
      </tbody>
    </table>
  );
};

export default ProductsList;
