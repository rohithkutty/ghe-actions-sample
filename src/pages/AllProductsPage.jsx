import React, { Component } from "react";
import ProductsList from "../components/ProductsList";

import "./AllProductsPage.styles.css";
import ProductsData from "../data/db.json";

export default class AllProductsPage extends Component {
  render() {
    const { products } = ProductsData;
    return (
      <div>
        <h2 className="products-header">Products List</h2>
        <ProductsList productsList={products} />
      </div>
    );
  }
}
