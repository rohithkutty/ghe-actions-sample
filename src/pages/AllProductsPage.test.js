import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import AllProductsPage from './AllProductsPage';
import ProductsList from '../components/ProductsList';
import ProductsData from '../data/db.json';

const { products } = ProductsData;
describe('AllProductsPage.jsx', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AllProductsPage />);
  });

  it('should render the correct heading', () => {
    expect(wrapper.find('h2').text()).toEqual('Products list');
  });
});

describe('When productsList array passed to ProductsList is null', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      productsList: null,
    };
    wrapper = shallow(<ProductsList {...props} />);
  });

  it('should not crash', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('When productsList array passed to ProductsList', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      productsList: products,
    };
    wrapper = shallow(<ProductsList {...props} />);
  });

  it('should not crash', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
