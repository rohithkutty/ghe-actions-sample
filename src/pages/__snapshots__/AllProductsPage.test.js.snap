// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When productsList array passed to ProductsList is null should not crash 1`] = `ShallowWrapper {}`;

exports[`When productsList array passed to ProductsList should not crash 1`] = `ShallowWrapper {}`;
